core/src/main/java/org/apache/struts2/dispatcher/multipart/JakartaStreamMultiPartRequest.java

```diff
@@ -52,309 +51,159 @@
+                    sanitizeNewlines(fileItemInput.getName()), maxSizeOfFiles);
         }
     }
 
     /**
-     * Processes the FileItemStream as a Form Field.
+     * Processes the FileItem as a file field.
      *
-     * @param itemStream file item stream
+     * @param fileItemInput file item representing upload file
+     * @param location      location
      */
-    protected void processFileItemStreamAsFormField(FileItemInput itemStream) {
-        String fieldName = itemStream.getFieldName();
-        try {
-            List<String> values;
-
-            String fieldValue = itemStream.getInputStream().toString();
-            if (!parameters.containsKey(fieldName)) {
-                values = new ArrayList<>();
-                parameters.put(fieldName, values);
-            } else {
-                values = parameters.get(fieldName);
-            }
-            values.add(fieldValue);
-        } catch (IOException e) {
-            LOG.warn("Failed to handle form field '{}'.", fieldName, e);
+    protected void processFileItemAsFileField(FileItemInput fileItemInput, Path location) throws IOException {
+        // Skip file uploads that don't have a file name - meaning that no file was selected.
+        if (fileItemInput.getName() == null || fileItemInput.getName().trim().isEmpty()) {
+            LOG.debug(() -> "No file has been uploaded for the field: " + sanitizeNewlines(fileItemInput.getFieldName()));
+            return;
         }
-    }
 
-    /**
-     * Processes the FileItemStream as a file field.
-     *
-     * @param itemStream file item stream
-     * @param location   location
-     */
-    protected void processFileItemStreamAsFileField(FileItemInput itemStream, String location) {
-        // Skip file uploads that don't have a file name - meaning that no file was selected.
-        if (itemStream.getName() == null || itemStream.getName().trim().isEmpty()) {
-            LOG.debug("No file has been uploaded for the field: {}", itemStream.getFieldName());
+        if (exceedsMaxFiles(fileItemInput)) {
             return;
         }
 
-        File file = null;
-        try {
-            // Create the temporary upload file.
-            file = createTemporaryFile(itemStream.getName(), location);
+        File file = createTemporaryFile(fileItemInput.getName(), location);
+        streamFileToDisk(fileItemInput, file);
 
-            if (streamFileToDisk(itemStream, file)) {
-                createFileInfoFromItemStream(itemStream, file);
-            }
-        } catch (IOException e) {
-            if (file != null) {
-                try {
-                    file.delete();
-                } catch (SecurityException se) {
-                    LOG.warn("Failed to delete '{}' due to security exception above.", file.getName(), se);
-                }
-            }
+        Long currentFilesSize = actualSizeOfUploadedFiles();
```

Comment:
This seems expensive to compute from scratch every time a file field is encountered, especially when the application might not have even configured a `maxSizeOfFiles` limit!

Reply:
Tbh I'm not convinced this option adds much value in the first place - I think `maxSize` serves this need well enough. I'm trying to think in what situation would an application need to configure `maxSizeOfFiles` instead of or in addition to `maxSize`. They want to accepts lots of big form fields but want to limit the amount of data written to disk? I guess it's possible

Reply:
> This seems expensive to compute from scratch every time a file field is encountered, especially when the application might not have even configured a `maxSizeOfFiles` limit!

Fixed!